fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -494,6 +501,56 @@ void testCloseOnCommitFailure() {
         }
     }
 
+    @Test
+    void testAutoCommitVersionStampOuterSleep() throws InterruptedException {
+        final byte[] key;
+        try (FDBRecordContext userContext = openContext()) {
+            key = this.path.toSubspace(userContext).pack(Tuple.from(prefix, "a").pack());
+            final AgilityContext agilityContext = AgilityContext.agile(userContext, 2, 10000);
+            AtomicReference<FDBRecordContext> firstOperation = new AtomicReference<>();
+            agilityContext.accept(context -> {
+                context.ensureActive().set(key, Tuple.from(1).pack());
+                firstOperation.set(context);
+            });
+            Thread.sleep(5);
+            // Here: after this operation, the first auto-context should be committed
```

Comment:
Is it worth asserting that the first context has not committed at this point, to prevent the possibility that the `accept` above takes 2ms?

Reply:
✅